balancer/xds/internal/regenerate_scripts/utils/proto-gen.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# packages is the collection of the packages that are required by xds for grpc.
+packages=(
+  envoy/service/discovery/v2:ads_go_grpc
+  envoy/api/v2:eds_go_grpc
+  envoy/api/v2:cds_go_grpc
+  envoy/api/v2/core:address_go_proto
+  envoy/api/v2/core:base_go_proto
+  envoy/api/v2/endpoint:endpoint_go_proto
+  envoy/type:percent_go_proto
+  envoy/service/load_stats/v2:lrs_go_grpc
+  udpa/data/orca/v1:orca_load_report_go_proto
+  udpa/service/orca/v1:orca_go_grpc
+)
+
+if [ -z $GOPATH ]; then echo 'empty $GOPATH, exiting.'; exit 1
+fi
+
+for i in ${packages[@]}
+do
+  bazel build "$i"
+done
+
+cwd=$(pwd)
+dest=$cwd/../../proto/
```

Comment:
Needs quotes in case the current directory has spaces.

Reply:
done